### PR TITLE
Code review edits

### PR DESCRIPTION
**Pull request by chrisc11**
- Move discussion of new features (DPLE, 2M & Coded PHY) 
   above "Auditing BLE Throughput Limiting Factors"
- Rename "Implementation Specific Details to Check" -> "Throughput Optimization Checklist" 
   and add an introductory blurb to the section
- Improve description in the "Optimizing throughput Example"
- minor grammatical fixes

### CODE REVIEW

File: _drafts/ble-throughput-primer.md [hunk 1]
```diff
@@ -141,6 +143,39 @@ Tabulating this information for a couple common MTU sizes we get:
 | 185 (iOS 10 max)    | 0.294             |
 | 512 (bluetooth max) | 0.301             |
 
+## Bluetooth Low Energy Specification Updates Impacting Throughput
```
Comment (chrisc11) [inline _drafts/ble-throughput-primer.md]: No real changes here, these lines were just moved

File: _drafts/ble-throughput-primer.md [hunk 2]
```diff
@@ -158,7 +193,13 @@ My two favorite analyzers on the market today are the
 
 Both of these are built on top of a _Software Defined Radio_ (_SDR_) meaning as the BLE spec evolves, the vendors should be able to ship a software update adding the support rather than requiring you purchase new hardware. Both these analyzers already support new features that have been added as part of BLE Specification revisions (Data Packet Length Extension, Channel Selection Algorithm #2, LE 2M & Coded PHY) which are becoming must have features when analyzing communications between the latest BLE chips. The Ellysis analyzer will also let you feed external GPIO lines into the analyzer that you can see synchronized with your transmissions sent over the air, which can be super helpful for debugging classes of bugs related to RF transmit windows.
 
-### Implementation Specific Details to Check
+### Throughput Optimization Checklist
```
Comment (chrisc11) [inline _drafts/ble-throughput-primer.md]: New blurb

File: _drafts/ble-throughput-primer.md [hunk 3]
```diff
@@ -213,40 +254,24 @@ When multiple devices are sharing the same radio, the software stack on the devi
...
+to be concerned about for BLE. This is because many many stacks drop data within the software
+stack. For example, BLE messages get queued up in the stack and when the heap holding the packets
+runs out of memory, some stacks will silently drop data. This means if you are sending large
+amounts of data over BLE you will usually want to add some sort of reliability layer that can
+detect & retransmit packets when data is dropped. The way this is implemented can have sizeable
+impacts on throughput. For example, if every single packet sent requires an ACK before another
+packet is sent, you'll typically wind up adding a connection interval worth of latency getting the
+ACK out, effectively halving the max throughput which can be achieved.
+
+#### Maximum Supported Link Layer Data Packet Size
```
Comment (chrisc11) [inline _drafts/ble-throughput-primer.md]: new section

**Comment by franc0is** [comment]
Ship it!

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Pull request merged**